src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -210,14 +210,23 @@ object Group {
   def defaultDependencies: Set[PathId] = Set.empty
   def defaultVersion: Timestamp = Timestamp.now()
 
-  implicit val groupValidator: Validator[Group] = validator[Group] { group =>
-    group.id is valid
-    group.apps is valid
-    group.groups is valid
-    group is noAppsAndGroupsWithSameName
-    (group.id.isRoot is false) or (group.dependencies is noCyclicDependencies(group))
-
-    group is validPorts
+  private def groupValidatorWithBase(base: PathId): Validator[Group] =
+    validator[Group] { group =>
+      group.id is pathAndBaseValidator(base)
+      group.apps is every(AppDefinition.nestedAppDefinitionValidator(base))
+      group is noAppsAndGroupsWithSameName
+      (group.id.isRoot is false) or (group.dependencies is noCyclicDependencies(group))
+      group is validPorts
+
+      group.dependencies is every(pathAndBaseValidator(base))
+      group.groups is every(valid(groupValidatorWithBase(base)))
+    }
+
+  implicit val groupValidator: Validator[Group] = new Validator[Group] {
+    override def apply(group: Group): Result = {
+      com.wix.accord.validate(group)(validator =
```

Comment:
Why the qualified call to validate?
Our initial idea to not import `com.wix.accord` did not work out, so we have 2 definitions of `validate`. We probably should either rename or remove the one in `Validation`.